### PR TITLE
SystemLink Enterprise Release Notes for 2024-10

### PR DESCRIPTION
**Pull request by robert-bob-1**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/install-systemlink-enterprise/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Release Notes for the 2024-10 release.

### Why should this Pull Request be merged?

Required for release.

### What testing has been done?

N/A

### CODE REVIEW

**Comment by CIakab-NI** [comment]
Hi @vpstoynova , could you help review the section `New Features and Behavior Changes'.
The info comes from this file: 
- [SLE 2024-10_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of September 2024.xls](https://nio365.sharepoint.com/:x:/s/ASWSystemLink/EXyFu9bNxqRCkAhWLAmogbQB4Nea-guya7p_zvz-h6D02Q?e=dJe91X)

I expect Mark to have technical questions, and have kept the content as squads have documented. Will tag Mark once I have your review.

Thanks!

File: release-notes/2024-10/README.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
+- testmonitorservice
```
Comment (CIakab-NI) [inline release-notes/2024-10/README.md:22]: @prestwick FYI, that this documentation is linked to this AzDO PR:
- [Pull Request 802072](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/802072): TestMonitor | Upgrade EF Core to 8.0

File: release-notes/2024-10/README.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
```
Comment (CIakab-NI) [inline release-notes/2024-10/README.md:24]: @prestwick FYI that this documentation is linked to this AzDO PR:
- [Pull Request 799320](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/799320): User Service | Enable CORS for delete session key endpoint

File: release-notes/2024-10/README.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
+        - Updated system maintainer to have artifact query privilege
+        - Update collaborator to have artifact query privilege
+- Notebook service - artifacts
```
Comment (CIakab-NI) [inline release-notes/2024-10/README.md:31]: @prestwick FYI that this documentation is linked to this AzDO PR:
- [Pull Request 789371](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/789371): NbExec | Artifacts | Update Authorization Actions

File: release-notes/2024-10/README.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+# SystemLink Enterprise release 2024-10 Release Notes
+
+The 2024-10 release for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.
+
+## Upgrading from the release 2024-09 to the release 2024-10
+
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
```
Comment (CIakab-NI) [inline release-notes/2024-10/README.md:11]: @prestwick , the squads have not documented any Helm Chart Breaking Changes (we can remove that section),
and they had only documented New Features and Behavior Changes.
Here's the doc:
- [SLE 2024-10_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of September 2024.xlsx](https://nio365.sharepoint.com/:x:/s/ASWSystemLink/EXyFu9bNxqRCkAhWLAmogbQB4Nea-guya7p_zvz-h6D02Q?e=TiL4Pa)

File: release-notes/2024-10/README.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+- Privileges and roles changes
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:11]:
```suggestion
- Changes to privileges and roles
```

File: release-notes/2024-10/README.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
+        - Updated system maintainer to have artifact query privilege
+        - Update collaborator to have artifact query privilege
+- Notebook service - artifacts
+    - Updated APIs(Create, Download & Update artifacts) to use new artifact actions
+- Notebook execution service
+    - Introduced new privilege for specifying resource profile: `notebookexecution:SpecifyResourceProfile`
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:32]:
```suggestion
    - Introduced new privilege for specifying a resource profile: `notebookexecution:SpecifyResourceProfile`
```

File: release-notes/2024-10/README.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
+        - Updated system maintainer to have artifact query privilege
+        - Update collaborator to have artifact query privilege
+- Notebook service - artifacts
+    - Updated APIs(Create, Download & Update artifacts) to use new artifact actions
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:30]:
```suggestion
    - Updated APIs (Create, Download & Update artifacts) to use new artifact actions.
```
Is the "&" part of the API name? If not, please use "and"

File: release-notes/2024-10/README.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
+- testmonitorservice
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
+        - Updated system maintainer to have artifact query privilege
+        - Update collaborator to have artifact query privilege
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:28]:
```suggestion
        - Update collaborator to have artifact query privilege.
```

File: release-notes/2024-10/README.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
+- testmonitorservice
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
+        - Updated system maintainer to have artifact query privilege
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:27]:
```suggestion
        - Updated system maintainer to have artifact query privilege.
```

File: release-notes/2024-10/README.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
+- testmonitorservice
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
+        - Updated data maintainer to have all artifact privileges
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:26]:
```suggestion
        - Updated data maintainer to have all artifact privileges.
```

File: release-notes/2024-10/README.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
+- testmonitorservice
+    - Ugpraded EF Core from 6.0 to 8.0, because the support for EF 6.0 ends on Nov 12, 2024.
+- userservice
+    - Enabled CORS for DELETE `niauth/v1/session-keys/self` route.
+- Service Registry & user service
+    - Created new actions for Artifact upload, download & update routes. Used in execution & dataspace privilege definition
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:25]:
```suggestion
    - Created new actions for artifact upload, download, and update routes. Used in execution & dataspace privilege definition
```
I don't understand this one. In the second sentence, what is used? The new actions?

File: release-notes/2024-10/README.md [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
+- Routines must now be disabled in order to be updated unless the user has all of enable, disable, and modify privileges.
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:19]:
```suggestion
- Users must disable routines to update them unless they have has all of enable, disable, and modify privileges.
```

File: release-notes/2024-10/README.md [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
+- As a result of the privilege changes above, non-admin users that could previously perform operations on Routines may not be able to any longer. These users should either be applied the "Routine Maintainer" role or have a new role created that grants Routine privileges.
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:18]:
```suggestion
- As a result of the privilege changes above, non-admin users that could previously perform operations on routines may not be able to any longer. Apply the "Routine Maintainer" role to these users or create a new role that grants routine privileges.
```

File: release-notes/2024-10/README.md [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
+    - "Notebook Execution" privilege category was renamed to "Scripts" to match the application rename
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:17]:
```suggestion
    - Renamed the "Notebook Execution" privilege category to "Scripts" to match the application rename.
```

File: release-notes/2024-10/README.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
+    - Routine "Modify metadata" privilege was renamed to "Modify routines"
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:16]:
```suggestion
    - Renamed the "Modify metadata" routine privilege to "Modify routines."
```

File: release-notes/2024-10/README.md [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
+    - "Cancel notebook execution" privilege was added. This isn't currently being enforced.
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:15]:
```suggestion
    - Added "Cancel notebook execution" privilege. This privilege is not currently enforced.
```

File: release-notes/2024-10/README.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+## Upgrading from the release 2024-09 to the release 2024-10
+
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
+    - Routines "Enable" and "Disable" privileges were added
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:14]:
```suggestion
    - Added "Enable" and "Disable" privileges to routines.
```

File: release-notes/2024-10/README.md [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+
+## Upgrading from the release 2024-09 to the release 2024-10
+
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
+    - "Routine Maintainer" role was added
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:13]:
```suggestion
    - Added "Routine Maintainer" role.
```

File: release-notes/2024-10/README.md [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+The 2024-10 release for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.
+
+## Upgrading from the release 2024-09 to the release 2024-10
+
+<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->
+
+## New Features and Behavior changes
+
+- Privileges and roles changes
+    - "Routines" service role was added
```
Comment (vpstoynova) [inline release-notes/2024-10/README.md:12]:
```suggestion
    - Added "Routines" service role.
```

**Review by vpstoynova — APPROVED** [review]

File: release-notes/2024-10/README.md (hunk 11 above)
Comment (prestwick) [reply release-notes/2024-10/README.md]: This capability doesn't actually appear to be shipping in this release. I have removed references to it.

File: release-notes/2024-10/README.md (hunk 12 above)
Comment (prestwick) [reply release-notes/2024-10/README.md]: Updated wording to align with suggestions.

File: release-notes/2024-10/README.md (hunk 13 above)
Comment (prestwick) [reply release-notes/2024-10/README.md]: Adjusted wording and capitalization based on recommendation.

File: release-notes/2024-10/README.md (hunk 14 above)
Comment (prestwick) [reply release-notes/2024-10/README.md]: Incorporating this suggestion.

**Review by prestwick — APPROVED** [review]

**Pull request merged**